### PR TITLE
Fix adding conformers with OpenMM units

### PR DESCRIPTION
**Pull request by mattwthompson**
In #1249 it was made obvious that I didn't test the case of adding a conformer with OpenMM units. The functionality was there but that particular test case was missing. I also found the code to be hard to follow and updated it in hopes of it being more maintainable.

- [x] Tag issue being addressed
- [x] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/master/openff/toolkit/tests)
- [x] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/master/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [x] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/master/docs/releasehistory.rst)

### CODE REVIEW

File: openff/toolkit/topology/molecule.py [hunk 1]
```diff
@@ -4246,50 +4246,56 @@ def _add_conformer(self, coordinates):
         index: int
             The index of this conformer
         """
-        new_conf = unit.Quantity(
-            np.zeros(shape=(self.n_atoms, 3), dtype=float), unit.angstrom
-        )
```
Comment (mattwthompson) [inline openff/toolkit/topology/molecule.py:4251]: I found this confusing - this `new_conf` read to me like the conformer passed through as an argument, but it's just a temporary variable used for  input validation. `coordinates` is the argument.

File: openff/toolkit/tests/test_molecule.py [hunk 2]
```diff
@@ -2492,6 +2491,8 @@ def test_equality(self):
 
     def test_add_conformers(self):
         """Test addition of conformers to a molecule"""
+        from openmm import unit as openmm_unit
+
```
Comment (j-wags) [inline openff/toolkit/tests/test_molecule.py:2495]: 👍 

At some point in the future we'll need to figure out how to handle test skips for openmm-free builds. But I don't think we need to handle that in this PR.

**Review by j-wags — APPROVED** [review]
Excellent. Thanks, @mattwthompson! Just one blocking comment, this looks great otherwise. Thanks for jumping on this so quickly and requesting my review.

File: openff/toolkit/tests/test_molecule.py (hunk 2 above)
Comment (mattwthompson) [reply openff/toolkit/tests/test_molecule.py:2495]: We should be able to get away with liberal use of [`skip_if_missing`](https://github.com/openforcefield/openff-utilities/blob/6db2421dec43f9e6e34ca38fa39f9bac86e00c85/openff/utilities/testing.py#L6) but I won't worry about that now

File: openff/toolkit/tests/test_molecule.py [hunk 3]
```diff
@@ -2622,6 +2622,26 @@ def test_add_conformers(self):
...
+            ),
+            openmm_unit.angstrom,
+        )
+        molecule.add_conformer(conf_openmm_units)
+
+        assert molecule.conformers[-1][0][0].m_as(unit.angstrom) == 1.0
+        assert molecule.conformers[-1][-1][-1].m_as(unit.angstrom) == 15.0
+
+        with pytest.raises(Exception) as excinfo:
+            molecule.add_conformer(conf_unitless)
```
[the remark this answers is not in the document]
Comment (mattwthompson) [reply openff/toolkit/tests/test_molecule.py:2643]: Wow, that's bad. You're right, and this is a side effect of the janky `with pytest.rasises(Exception):` which doesn't really do what we want.

[the remark this answers is not in the document]
Comment (mattwthompson) [reply openff/toolkit/tests/test_molecule.py:2643]: I'll resist the temptation to fix this in _this_ PR, but:

```shell
$ grep -r "pytest.raises(Exception)" openff/toolkit/tests | wc -l
      15
